ctapipe/io/eventfilereader.py

```diff
@@ -2,36 +2,83 @@
 Handles reading of different event/waveform containing files
 """
 from abc import abstractmethod
-from os.path import basename, splitext, dirname, join, exists
-import numpy as np
-from traitlets import Unicode, Int, CaselessStrEnum, observe
+from os.path import exists
+from traitlets import Unicode, Int, CaselessStrEnum
 from copy import deepcopy
 from ctapipe.core import Component, Factory
 from ctapipe.utils import get_dataset
-from ctapipe.io.hessio import hessio_event_source, hessio_get_list_event_ids
+from ctapipe.core import Provenance
 
 
 class EventFileReader(Component):
     """
-    Parent class for EventFileReaders of different sources. A new
-    EventFileReader should be created for each type of event file read into
-    ctapipe, e.g. simtelarray files are read by the `HessioFileReader`.
+    Parent class for EventFileReaders of different sources.
+
+    A new EventFileReader should be created for each type of event file read
+    into ctapipe, e.g. sim_telarray files are read by the `HessioFileReader`.
+
+    EventFileReader provides a common high-level interface for accessing event
+    data from different data sources. Creating an EventFileReader for a new
+    data source ensures that data can be accessed in a common way,
+    irregardless of the data source.
+
+    EventFileReader itself is an abstract class. To use an EventFileReader you
+    must use a subclass that is relevant for the data format for the file you
+    are reader (for example you must use
+    `ctapipe.io.hessiofilereader.HessioFileReader` to read a hessio format
+    file). Alternatively you can use
+    `ctapipe.io.eventfilereader.EventFileReaderFactory` to automatically
+    select the correct EventFileReader subclass for the data format you wish
+    to read.
+
+    To loop through the events in a file:
+
+    >>> reader = EventFileReader(None, None, input_path=path)
```

Comment:
Actually the correct way would be to have this Component initialised from inside a `ctapipe.core.tool.Tool`. In which case it would be: 
```
foo = SomeClass(self.config, self)
```
I was going to write that in the docstring, but I was worried that would be more confusing. I will change it, and point the reader to an example script using it. Does that sound better?

Reply:
I agree- that's a refactoring for another PR though... I don't like the "None, None", but it comes from the underlying `traitlets.config` code (at least the first one, the second one is my fault and can be removed, since i don't think we need to keep a pointer to the parent anywhere anymore).  In traitlets, the first None tels the class not to try to read the config from the global configuration file or `sys.argv`, so usually you don't need it unless you're in a notebook or script where you are not inside a `Tool` base class.